### PR TITLE
feat: validate eligibility flag as per record type

### PR DESCRIPTION
**Pull request by sudipnhs**
## Type of changes

- [ ] Refactoring (non-breaking change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would change existing functionality)
- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [ ] I am familiar with the [contributing guidelines](../docs/CONTRIBUTING.md)
- [x] I have followed the code style of the project
- [ ] I have added tests to cover my changes
- [ ] I have updated the documentation accordingly
- [ ] This PR is a result of pair or mob programming

---

## Sensitive Information Declaration

To ensure the utmost confidentiality and protect your and others privacy, we kindly ask you to NOT including [PII (Personal Identifiable Information) / PID (Personal Identifiable Data)](https://digital.nhs.uk/data-and-information/keeping-data-safe-and-benefitting-the-public) or any other sensitive data in this PR (Pull Request) and the codebase changes. We will remove any PR that do contain any sensitive information. We really appreciate your cooperation in this matter.

- [x] I confirm that neither PII/PID nor sensitive data are included in this PR and the codebase changes.

### CODE REVIEW

File: application/CohortManager/rules/Breast_Screening_staticRules.json [hunk 1]
```diff
@@ -87,6 +87,36 @@
...
+          {
+            "Name": "delRecordType",
+            "Expression": "participant.RecordType == Actions.Removed"
+          },
+          {
+            "Name": "validDelEligibilityFlag",
+            "Expression": "participant.EligibilityFlag == true"
+          }
+        ],
+        "Expression": "(newRecordType AND validNewEligibilityFlag) OR (newRecordType AND validDelEligibilityFlag)",
```
Comment (andrewmorris43) [inline application/CohortManager/rules/Breast_Screening_staticRules.json:111]: Should the second condition be:
(delRecordType AND validDelEligibilityFlag)

Or do you not need the delRecordType param?

File: tests/ScreeningValidationServiceTests/StaticValidation/StaticValidationTests.cs [hunk 2]
```diff
@@ -1035,4 +1035,27 @@ public async Task Run_CurrentPostingAndPrimaryCareProvider_DoesNotCreateExceptio
             Times.Never());
     }
     #endregion
+
+    #region Validate Eligibility Flag as per Record Type (Rule 94)
+    [TestMethod]
```
Comment (andrewmorris43) [inline tests/ScreeningValidationServiceTests/StaticValidation/StaticValidationTests.cs:1040]: Could we also have a unit test for when the exception should not be raised.

File: application/CohortManager/rules/Breast_Screening_staticRules.json (hunk 1 above)
Comment (sudipnhs) [reply application/CohortManager/rules/Breast_Screening_staticRules.json]: Andrew, you are right!. It was a silly mistake from my side. I have now rectified it.
Logic behind this is as follows-
For recordtype = "ADD", eligibility flag cannot be false
For recordtype = "DEL", eligibility flag cannot be true

Comment (sudipnhs) [reply application/CohortManager/rules/Breast_Screening_staticRules.json]: Andrew, you are right!. It was a silly mistake from my side. I have now rectified it.
Logic behind this is as follows-
For recordtype = "ADD", eligibility flag cannot be false
For recordtype = "DEL", eligibility flag cannot be true

Comment (andrewmorris43) [reply application/CohortManager/rules/Breast_Screening_staticRules.json]: Easily done, I've made plenty of these type of mistakes working on various rules!

**Pull request closed without merging**